native_client/javascript/index.ts

```diff
@@ -0,0 +1,288 @@
+import binary from 'node-pre-gyp';
+import path from 'path';
+
+// 'lib', 'binding', 'v0.1.1', ['node', 'v' + process.versions.modules, process.platform, process.arch].join('-'), 'deepspeech-bindings.node')
+const binding_path = binary.find(path.resolve(path.join(__dirname, 'package.json')));
+
+// On Windows, we can't rely on RPATH being set to $ORIGIN/../ or on
+// @loader_path/../ but we can change the PATH to include the proper directory
+// for the dynamic linker
+if (process.platform === 'win32') {
+    const dslib_path = path.resolve(path.join(binding_path, '../..'));
+    var oldPath = process.env.PATH;
+    process.env['PATH'] = `${dslib_path};${process.env.PATH}`;
+}
+
+const binding = require(binding_path);
+
+if (process.platform === 'win32') {
+  process.env['PATH'] = oldPath;
+}
+
+/**
+ * Stores text of an individual token, along with its timing information
+ */
+export interface TokenMetadata {
+    /** The text corresponding to this token */
+    text: string;
+
+    /** Position of the token in units of 20ms */
+    timestep: number;
+
+    /** Position of the token in seconds */
+    start_time: number;
+}
+
+/**
+ * A single transcript computed by the model, including a confidence value and
+ * the metadata for its constituent tokens.
+ */
+export interface CandidateTranscript {
+    tokens: TokenMetadata[];
+
+    /**
+     * Approximated confidence value for this transcription. This is roughly the
+     * sum of the acoustic model logit values for each timestep/token that
+     * contributed to the creation of this transcription.
+     */
+    confidence: number;
+}
+
+/**
+ * An array of CandidateTranscript objects computed by the model.
+ */
+export interface Metadata {
+    transcripts: CandidateTranscript[];
+}
+
+/**
+ * Provides an interface to a DeepSpeech stream. The constructor cannot be called
+ * directly, use :js:func:`Model.createStream`.
+ */
+export class Stream {
```

Comment:
Thinking more about it. I think this should be just `class Stream`. The constructor is private so we don't want to export it, `Model.createStream` should be the only way to get a stream.